domains/client/domain-operator/src/bundle_processor.rs

```diff
@@ -65,6 +71,60 @@ where
     }
 }
 
+struct ReceiptValidator<Client> {
+    client: Arc<Client>,
+}
+
+impl<Client> Clone for ReceiptValidator<Client> {
+    fn clone(&self) -> Self {
+        Self {
+            client: self.client.clone(),
+        }
+    }
+}
+
+impl<Client> ReceiptValidator<Client> {
+    pub fn new(client: Arc<Client>) -> Self {
+        Self { client }
+    }
+}
+
+impl<Block, CBlock, Client> domain_block_preprocessor::ValidateReceipt<Block, CBlock>
+    for ReceiptValidator<Client>
+where
+    Block: BlockT,
+    CBlock: BlockT,
+    Client: AuxStore,
+{
+    fn validate_receipt(
+        &self,
+        receipt: &ExecutionReceiptFor<Block, CBlock>,
+    ) -> sp_blockchain::Result<ReceiptValidity> {
+        // Skip genesis receipt as it has been already verified by the consensus chain.
+        if receipt.domain_block_number.is_zero() {
+            return Ok(ReceiptValidity::Valid);
+        }
+
+        let consensus_block_hash = receipt.consensus_block_hash;
+        let local_receipt = crate::aux_schema::load_execution_receipt::<_, Block, CBlock>(
```

Comment:
do not check anything other than invalid bundles for the receipt check or a TODO for later PR ?

Reply:
Yeah, the other things are left as TODOs for the follow-up as many of them are not yet tracked in the receipt. Basically, we need to introduce and check all the domain state-related fields in the receipt: domain_block_extrinsics_root, post_state_root, rewards_info, XDM, every time we add a new  field, the invalid_bundles will be extended as well.